ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/Endpoint.java

```diff
@@ -49,15 +52,15 @@ public Endpoint(final String host, final int udpPort, final Optional<Integer> tc
   }
 
   public static Endpoint fromEnode(final EnodeURL enode) {
-    final int discoveryPort =
-        enode
-            .getDiscoveryPort()
-            .orElseThrow(
-                () ->
-                    new IllegalArgumentException(
-                        "Attempt to create a discovery endpoint for an enode with discovery disabled."));
+    Optional<Integer> discoveryPort = enode.getDiscoveryPort();
+
+    if (discoveryPort.isEmpty()) {
+      log.warn("Attempted to create a discovery endpoint for a node with discovery disabled: {}", enode);
```

Comment:
if the discovery port is empty, the default port will be used, right? Can we add a test for the expected outcome?

And in that case, should this log be moved to debug? "Using default port for enode"

Reply:
@macfarla Understood! What would be a good location for this test?